liblustreapi-sys/build.rs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2019 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use std::path::PathBuf;
+
+fn main() {
+    let out_file = PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+        .join("src")
+        .join("bindings.rs");
+
+    // Tell cargo to tell rustc to link the system liblustreapi
+    // shared library.
+    println!("cargo:rustc-link-lib=lustreapi");
+
+    if out_file.exists() {
```

Comment:
This return should be before the println!

Reply:
Nope.  If that happens, it doesn't link correctly:
```$ cargo build
   Compiling liblustreapi-sys v0.1.0 (/home/nclark/integrated-manager-lustre/liblustreapi-sys)
   Compiling liblustreapi v0.1.0 (/home/nclark/integrated-manager-lustre/liblustreapi)
   Compiling stratagem-runner v0.1.0 (/home/nclark/integrated-manager-lustre/stratagem-runner)
error: linking with `cc` failed: exit code: 1
  |
  = note: "cc" "-Wl,--as-needed" "-Wl,-z,noexecstack" "-m64" "-L" "/usr/lib/rustlib/x86_64-unknown-linux-gnu/lib" "/home/nclark/integrated-manager-lustre/target/debug/deps/stratagem_purge-8d9a2b6204dca4d4.12kla9cy65x1yhjz.rcgu.o" "/home/nclark/integrated-manager-lustre/target/debug/deps/stratagem_purge-8d9a2b6204dca4d4.1bb95z8b7secl8aw.rcgu.o" "/home/nclark/integrated-manager-lustre/target/debug/deps/stratagem_purge-8d9a2b6204dca4d4.1i2vrvvrgz2hhpp8.rcgu.o" "/home/nclark/integrated-manager-lustre/target/debug/deps/stratagem_purge-8d9a2b6204dca4d4.1pde6m5quu76cyek.rcgu.o" "/home/nclark/integrated-manager-lustre/target/debug/deps/stratagem_purge-8d9a2b6204dca4d4.1xusjp2ruj5tei6a.rcgu.o" "/home/nclark/integrated-manager-lustre/target/debug/deps/stratagem_purge-8d9a2b6204dca4d4.2lo5emoto6tcqzjo.rcgu.o" "/home/nclark/integrated-manager-lustre/target/debug/deps/stratagem_purge-8d9a2b6204dca4d4.2qiobr7i5e3jw5gi.rcgu.o" "/home/nclark/integrated-manager-lustre/target/debug/deps/stratagem_purge-8d9a2b6204dca4d4.321x5b42melgvdba.rcgu.o" "/home/nclark/integrated-manager-lustre/target/debug/deps/stratagem_purge-8d9a2b6204dca4d4.3323k4t45b4gj399.rcgu.o" "/home/nclark/integrated-manager-lustre/target/debug/deps/stratagem_purge-8d9a2b6204dca4d4.3c3erb7g09qz4ley.rcgu.o" "/home/nclark/integrated-manager-lustre/target/debug/deps/stratagem_purge-8d9a2b6204dca4d4.3dmd75i0hchd49d6.rcgu.o" "/home/nclark/integrated-manager-lustre/target/debug/deps/stratagem_purge-8d9a2b6204dca4d4.3e4qndibgd7c01s5.rcgu.o" "/home/nclark/integrated-manager-lustre/target/debug/deps/stratagem_purge-8d9a2b6204dca4d4.3m8eixizit3upxps.rcgu.o" "/home/nclark/integrated-manager-lustre/target/debug/deps/stratagem_purge-8d9a2b6204dca4d4.3yzpe4qsz1ml217q.rcgu.o" "/home/nclark/integrated-manager-lustre/target/debug/deps/stratagem_purge-8d9a2b6204dca4d4.4a7404x5vbjnftwb.rcgu.o" "/home/nclark/integrated-manager-lustre/target/debug/deps/stratagem_purge-8d9a2b6204dca4d4.4bhhf4muakdgaaia.rcgu.o" "/home/nclark/integrated-manager-lustre/target/debug/deps/stratagem_purge-8d9a2b6204dca4d4.4par896ruacl8cpq.rcgu.o" "/home/nclark/integrated-manager-lustre/target/debug/deps/stratagem_purge-8d9a2b6204dca4d4.4uxggcsc2qkdwd36.rcgu.o" "/home/nclark/integrated-manager-lustre/target/debug/deps/stratagem_purge-8d9a2b6204dca4d4.54sde2aehb5z5th6.rcgu.o" "/home/nclark/integrated-manager-lustre/target/debug/deps/stratagem_purge-8d9a2b6204dca4d4.59vsmzkf5zfn0fv7.rcgu.o" "/home/nclark/integrated-manager-lustre/target/debug/deps/stratagem_purge-8d9a2b6204dca4d4.5bwz3kpsenuy2ks5.rcgu.o" "/home/nclark/integrated-manager-lustre/target/debug/deps/stratagem_purge-8d9a2b6204dca4d4.8ds7ev4t7qvds71.rcgu.o" "/home/nclark/integrated-manager-lustre/target/debug/deps/stratagem_purge-8d9a2b6204dca4d4.dk93sk1demxt5ms.rcgu.o" "/home/nclark/integrated-manager-lustre/target/debug/deps/stratagem_purge-8d9a2b6204dca4d4.t5yoibj9w9y38ys.rcgu.o" "/home/nclark/integrated-manager-lustre/target/debug/deps/stratagem_purge-8d9a2b6204dca4d4.tgbbtu59n6xop8s.rcgu.o" "/home/nclark/integrated-manager-lustre/target/debug/deps/stratagem_purge-8d9a2b6204dca4d4.u1zgqzm6avk7yok.rcgu.o" "-o" "/home/nclark/integrated-manager-lustre/target/debug/deps/stratagem_purge-8d9a2b6204dca4d4" "/home/nclark/integrated-manager-lustre/target/debug/deps/stratagem_purge-8d9a2b6204dca4d4.12lu82m7lxh0biq8.rcgu.o" "-Wl,--gc-sections" "-pie" "-Wl,-zrelro" "-Wl,-znow" "-nodefaultlibs" "-L" "/home/nclark/integrated-manager-lustre/target/debug/deps" "-L" "/usr/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-Wl,-Bstatic" "/home/nclark/integrated-manager-lustre/target/debug/deps/libliblustreapi-c9b4156e4beaff68.rlib" "/home/nclark/integrated-manager-lustre/target/debug/deps/libliblustreapi_sys-8312c18576d5bd5e.rlib" "/home/nclark/integrated-manager-lustre/target/debug/deps/liberrno-50bc5fdf875d2ff8.rlib" "/home/nclark/integrated-manager-lustre/target/debug/deps/liblibc-711591a738fe4507.rlib" "-Wl,--start-group" "/usr/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-0e272be2569672f0.rlib" "/usr/lib/rustlib/x86_64-unknown-linux-gnu/lib/libpanic_unwind-369c8b15956ca76c.rlib" "/usr/lib/rustlib/x86_64-unknown-linux-gnu/lib/libbacktrace_sys-3c6f04eb2b03d2b3.rlib" "/usr/lib/rustlib/x86_64-unknown-linux-gnu/lib/libunwind-c56f1da838c839da.rlib" "/usr/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_demangle-7a93a39a107d6ec4.rlib" "/usr/lib/rustlib/x86_64-unknown-linux-gnu/lib/liblibc-1bae7a39c5eff233.rlib" "/usr/lib/rustlib/x86_64-unknown-linux-gnu/lib/liballoc-f65a4b83daa5ffd7.rlib" "/usr/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_std_workspace_core-c56dc9794e055f14.rlib" "/usr/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcore-139abab598e46a29.rlib" "-Wl,--end-group" "/usr/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-806a4ba0b6854dbf.rlib" "-Wl,-Bdynamic" "-lutil" "-lutil" "-ldl" "-lrt" "-lpthread" "-lgcc_s" "-lc" "-lm" "-lrt" "-lpthread" "-lutil" "-lutil"
  = note: /home/nclark/integrated-manager-lustre/target/debug/deps/libliblustreapi-c9b4156e4beaff68.rlib(liblustreapi-c9b4156e4beaff68.15a6bl2hegcj2r7s.rcgu.o): In function `liblustreapi::fid2path::h1bff9ea0b0268d7c':
          /home/nclark/integrated-manager-lustre/liblustreapi/src/lib.rs:95: undefined reference to `llapi_fid2path'
          /home/nclark/integrated-manager-lustre/target/debug/deps/libliblustreapi-c9b4156e4beaff68.rlib(liblustreapi-c9b4156e4beaff68.15a6bl2hegcj2r7s.rcgu.o): In function `liblustreapi::search_rootpath::hae2c5ea8ae94479d':
          /home/nclark/integrated-manager-lustre/liblustreapi/src/lib.rs:126: undefined reference to `llapi_search_rootpath'
          collect2: error: ld returned 1 exit status
          

error: aborting due to previous error
```

Reply:
hrrm, but if it's not there compilation won't happen at all on mac as it's put in the linker